lib/rules/no-implicit-injections.js

```diff
@@ -0,0 +1,177 @@
+'use strict';
+
+const emberUtils = require('../utils/ember');
+const { getImportIdentifier } = require('../utils/import');
+const Stack = require('../utils/stack');
+const types = require('../utils/types');
+const kebabCase = require('lodash.kebabcase');
+const camelCase = require('lodash.camelcase');
+
+const defaultServiceConfig = { serviceName: 'store', moduleNames: ['Route', 'Controller'] };
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+function fixService(fixer, currentClass, serviceInjectImportName, servicePropertyName) {
+  const serviceInjectPath = kebabCase(servicePropertyName);
+
+  return fixer.insertTextBefore(
+    currentClass.node.body.body[0],
+    `@${serviceInjectImportName}('${serviceInjectPath}') ${servicePropertyName};\n`
+  );
+}
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'enforce usage of implicit service injections',
+      category: 'Deprecations',
+      recommended: false,
+      url: 'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-implicit-injections.md',
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          services: {
+            type: 'array',
+            items: {
+              type: 'object',
+              default: [{ serviceName: 'store', moduleNames: ['Route', 'Controller'] }],
+              properties: {
+                serviceName: {
+                  type: 'string',
+                },
+                moduleNames: {
+                  type: 'array',
+                  items: {
+                    type: 'string',
+                  },
+                },
+              },
+            },
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+    messages: {
+      main: 'Do not rely on implicit service injections for the "{{serviceName}}" service. Implicit service injections were deprecated in Ember 3.26 and will not work in 4.0.',
+    },
+  },
+
+  create(context) {
+    const options = context.options[0] || {
+      services: [defaultServiceConfig],
+    };
+    const servicesConfig = options.services || [defaultServiceConfig];
+
+    // State being tracked for this file.
+    let serviceInjectImportName = undefined;
+    let serviceNamesWithImplicitInjection = servicesConfig;
+    let isEmberModule = false;
+
+    // State being tracked for the current class we're inside.
+    const classStack = new Stack();
+
+    // This rule does not apply to test files or non class modules
+    if (emberUtils.isTestFile(context.getFilename())) {
```

Comment:
According to test coverage checks, this line is not tested. Can you test it? And also be sure to monitor the test coverage to ensure every line in the new rule is covered.

```
yarn test --coverage lib/rules/no-deprecated-router-transition-methods.js
```

Reply:
I'm rethinking the need to not run this in tests. You could have mock services, routes, controllers, components, etc declared in tests that would still suffer from the same deprecation and we'd want to catch those.